src/App.js

```diff
@@ -1,31 +1,106 @@
+import React, { useState, Component } from 'react';
 import { makeStyles } from '@material-ui/core/styles';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
+import Card from '@material-ui/core/Card';
+import Button from '@material-ui/core/Button';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import TextField from '@material-ui/core/TextField';
+import ROSLIB from 'roslib';
 
-const useStyles = makeStyles((theme) => ({
-  root: {
-    flexGrow: 1
-  },
-  title: {
-    flexGrow: 1
+const ros = new ROSLIB.Ros();
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: 'ws://localhost:9090',
+      connected: false,
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleConnect = this.handleConnect.bind(this);
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+    this.setState({value: event.target.value});
+  }
+
+  handleConnect(event) {
+    event.preventDefault();
+    ros.connect(this.state.value);
+
+    ros.on('connection', () => {
+      this.setConnected(true);
+    });
+  
+    ros.on('error', () => {
+      this.setConnected(false);
+      alert('error');
+    });
+  
+    ros.on('close', () => {
+      this.setConnected(false);
+      alert('close');
+    });
+  }
+
+  setConnected(bool) {
```

Comment:
Use a `useState()` hooks, so you won't need to create a `setConnected()` function.